xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/internal/html/DefaultHTMLCleaner.java

```diff
@@ -221,10 +222,8 @@ private CleanerProperties getDefaultCleanerProperties(HTMLCleanerConfiguration c
         // TODO: handle HTML5 correctly (see: https://jira.xwiki.org/browse/XCOMMONS-901)
         defaultProperties.setHtmlVersion(4);
 
-        // Don't trim spaces in attribute values
-        // We don't want input such as: <input type="hidden" value="  foo" />
-        // to be modified for: value="foo"
-        // if such case happen, the value should remain the same.
+        // We don't trim any attribute value by default, so we can keep the leading space in input value attributes.
```

Comment:
> We don't trim any attribute value by default

I think it's the opposite. We want to trim attribute values by default except for HTML INPUT tag attributes. Right?

Reply:
Yep, that is what we want. But here I explain why we deactivate the flag: we don't let HtmlCleaner trim the attribute. We do it ourselves through the transformation.

Reply:
ok I understand now but it wasn't very clear. I thought you were explaining what we want (which is what I would expect in such a comment). Explain what we want and then explain how to get there.

Reply:
I updated the comment.